salt/states/pkgrepo.py

```diff
@@ -387,6 +402,14 @@ def absent(name, **kwargs):
         if not name.startswith('ppa:'):
             name = 'ppa:' + name
 
+    remove_key = any(kwargs.get(x) is not None
+                     for x in ('keyid', 'keyid_ppa'))
```

Comment:
@terminalmage that means that if both `keyid` and `keyid_ppa` is `False`, remove_key will still be True?

I think it should be:

```python
remove_key = any(kwargs.get(x, False)
                               for x in ('keyid', 'keyid_ppa'))
```

haven't tested, though.